app/assets/javascripts/transitive_dependencies.js

```diff
@@ -0,0 +1,85 @@
+function ajaxDepOnClick(gem_id,ver_id) {
+  $.ajax({
+    type: "get",
+    url: "/gems/"+gem_id+"/versions/"+ver_id+"/transitive_dependencies.json",
+    success: function(resp) {
+      //var json_obj = JSON.parse(resp);
+      //alert(JSON.stringify(resp))
+      ajaxDepActions(resp)
+    }
+  });
+}
+
+function ajaxDepActions(resp) {
+  var gem_id = resp.rubygem_id
+
+  // when we have gems like http_parser.rb, we need to replace . with \. to access any ids with the name
+  gem_id = gem_id.replace(/\./, "\\.")
+
+  var all_deps = $('.deps_item').map(function() {
+    return $(this).attr('data-id');
+  }).get();
+
+  if (resp.run_deps.length != 0){
+    $("#"+gem_id+'runtime').find(".deps_scope").before("<span class='scope scope--expanded'>Runtime :</span>");
+    deps_display(resp.run_deps,gem_id,all_deps,"runtime")
+  }
+
+  if (resp.dev_deps.length != 0){
+    $("#"+gem_id+'development').find(".deps_scope").before("<span class='scope'>Development :</span>");
+    deps_display(resp.dev_deps,gem_id,all_deps,"development")
+  }
+
+  var toggler = "<span class='deps_expanded deps_expanded-down'></span>";
+  $('#'+gem_id+'-toggle').html(toggler);
+  $('#'+gem_id+'-toggle').click(function() {
+    $(this).parent().parent().find('div').first().toggleClass('deps_toggle');
+    $(this).parent().parent().find('div').first().next().toggleClass('deps_toggle');
+    $(this).find('span').first().toggleClass('deps_expanded-down');
+  });
+
+  $('#'+gem_id+'-title').addClass('deps_item--title')
+  $('.deps_item--duplicate').css("padding", "20")
+}
+
+function deps_display(deps_names,gem_id,all_deps,scope) {
+  var new_gems = $("#"+gem_id+scope).find(".deps_scope")
+  var duplicate_gems = $("#"+gem_id+scope).find(".dup_scope")
+  $.each(deps_names, function (idx,dep_details) {
+    dep = dep_details[0]
+    ver_num = dep_details[1]
+    req = dep_details[2]
+    if(all_deps.includes(dep)) {
```

Comment:
In the following case, rake 11.3.0 was not toggleable as we have shown rake 12.3.2, which isn't right. 12.3.2 could have different dependencies than 11.3.0
![Screenshot from 2019-08-16 00-08-12](https://user-images.githubusercontent.com/7680662/63117765-13e35a80-bfba-11e9-83e6-43f2418b733f.png)

This condition of already shown is making everything complicated. I think we should remove this check and always allow further toggle.

Reply:
Okay. I'll change that or I can check name and version. Not a lot of change in code.